azure_functions_worker/bindings/meta.py

```diff
@@ -25,12 +30,30 @@ def load_binding_registry() -> None:
     global BINDING_REGISTRY
     BINDING_REGISTRY = func.get_binding_registry()
 
-
-def get_binding(bind_name: str) -> object:
-    binding = None
-    registry = BINDING_REGISTRY
-    if registry is not None:
-        binding = registry.get(bind_name)
+    # The SDKs only support python 3.9+
+    if sys.version_info.minor >= BASE_EXT_SUPPORTED_PY_VERSION:
+        # Check if cx has imported sdk bindings library
+        try:
+            clients = importlib.util.find_spec('azure.functions.extension.base')
+        except ModuleNotFoundError:
+            # This will throw a ModuleNotFoundError in env reload because
+            # azure.functions.extension isn't loaded in
+            clients = None
+
+        # This will be None if the library is not imported
+        # If it is not None, we want to set and use the registry
+        if clients is not None:
+            import azure.functions.extension.base as clients
+            global SDK_BINDING_REGISTRY
+            SDK_BINDING_REGISTRY = clients.get_binding_registry()
+
+
+def get_binding(bind_name: str, pytype: typing.Optional[type] = None) -> object:
+    binding = get_deferred_binding(bind_name=bind_name, pytype=pytype)
+
+    # Either cx didn't import library or didn't define sdk type
+    if binding is None and BINDING_REGISTRY is not None:
+        binding = BINDING_REGISTRY.get(bind_name)
     if binding is None:
         binding = generic.GenericBinding
```

Comment:
Fixed

Reply:
There isn't a clean way to improve this check. For generic bindings, `binding = BINDING_REGISTRY.get(bind_name)` will return None. We still need to check if binding is None after this call, otherwise the binding won't be set to `generic.GenericBinding` and the function will return None.